### DISCUSSION TITLE
Changing fp_size parameter recommendations

### DISCUSSION THREAD

**Discussion by billramo** [rdkit/rdkit#6644] [Q&A] [answered]
I'm work on the Amazon RDS/Aurora team. When we first added the rdkit extension to RDS and Aurora PostgreSQL three years ago, Greg Landrum recommended to our development team to lock down the ability for customers to change the fp_size parameters. RDS/Aurora PostgreSQL only supports customers to change 'rdkit.dice_threshold', 'rdkit.do_chiral_sss', and 'rdkit.tanimoto_threshold'.

Is this still the recommendation for users of rdkit? Is there a way for customers to use the API to set for example the rdkit.morgan_fp_size without using the parameter. If so, are there any consequences for customers who make the change like needed to rebuild all related indexes? There is an open issue https://github.com/rdkit/rdkit/issues/5583 that really should be targeted to our team related to morgan_fp_size. 
Thanks,
Bill Ramos, Senior Product Manager Technical, Amazon Web Services

**Comment by greglandrum** [comment, marked as answer]
Hi @billramo,
At the moment the only control over fingerprint size in the cartridge is by using the parameter. 

This is a tricky issue in the cartridge since the code requires the fingerprints used for similarity comparison (and building any indices) to be the same length. When I talked to the team about disabling the parameter, the idea was that it's too easy for users to make a mistake and to only change the parameter before some calls. I think the idea was "idiot-proofing" things to a certain extent.

It's always possible to change this decision and allow user access (while accepting the ensuing risks).

One thing that might be possible: if PostgreSQL and/or Aurora support persistent changes of these variables at the database level then you could imagine allowing admins to set the parameters for an entire database at a time. If this is possible (I don't know the systems well enough) it would solve the consistency problem.

-greg

**Comment by greensmith** [comment]
@billramo  @greglandrum 

How come it was concluded that idiot proofing was the right choice?  Surely it is the end-user who is responsible for the provision of the RDS db  and  for the data within it; if the customer is changing this value that is their risk?

Secondly as far as I am to understand setting AWS RDS parameters is not usually something you want to do "on the fly" or "change the parameter before some calls"?  If the parameter is defined as static the db must be restarted for the change to take effect.   Is it not simply the case that you could mark the RDKIT parameters as static and require reboots to the DB for those changes to take effect?

edit: to add some context, we were looking to change the rdkit_fp_size permanently from its default, and are not planning on changing it willy-nilly
